### PR TITLE
fix: menu dropdown that do not close properly

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: regen-network/regen-registry#1689

- did some refactoring of some header related components (the previous structure/names were a bit hard to follow IMO)
- see this commit: ~~94d44cbbe3645e1ef90bb5817c5595e0b4b6fb20~~ https://github.com/regen-network/regen-web/pull/1912/commits/2a4d9d145964217c2308f74aecf862129ed6fdb7 for the actual fix

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

https://deploy-preview-1912--regen-registry.netlify.app/

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by wgwz** [comment]
@blushi could you explain the fix commit a bit? i'm curious to understand it better

File: web-components/src/components/header/components/HeaderMenuItem/HeaderMenuItem.Hover.tsx [hunk 1]
```diff
@@ -67,43 +67,41 @@ const MenuHover = ({
         )}
         {renderTitle && renderTitle()}
       </span>
-      {popoverAnchor.current !== null && (
```
Comment (flagrede) [inline web-components/src/components/header/components/HeaderMenuItem/HeaderMenuItem.Hover.tsx:70]: I remember doing that to prevent the dropdown to appears in the bottom right screen corner but apparently it created a new issue. I could no longer reproduce the original issue on the deploy preview so I can guess it's safe to remove now.

**Review by flagrede — APPROVED** [review]
Nice refactor!
Regarding the fix not sure why the original issue describe [here](https://github.com/regen-network/regen-web/pull/1709) is gone but it's a good thing anyway.

One last thing, not directly concerning PR, I noticed that the dropdown menu is triggered only when hovering over the user icon and not the button. Would be nice to have the hover zone on the button if it's easy to do, otherwise, we can do that in a separate ticket.
image

File: web-components/src/components/header/components/HeaderMenuItem/HeaderMenuItem.Hover.tsx (hunk 1 above)
Comment (blushi) [reply web-components/src/components/header/components/HeaderMenuItem/HeaderMenuItem.Hover.tsx:70]: oh actually just realized it does appear back again at the bottom right, but then disappears very quickly, which makes it hard to catch! so I need to look into this a bit closer...

Comment (blushi) [reply web-components/src/components/header/components/HeaderMenuItem/HeaderMenuItem.Hover.tsx]: so looking more closely at some mui example: https://mui.com/material-ui/react-popover/#mouse-over-interaction and  also react docs regarding documentation for using refs (in particular do not read while rendering, see "Pitfall" in https://react.dev/reference/react/useRef#usage), I switched to using state but still need to keep the ref to the menu item title so the popover is still displayed below it once we hover on the dropdown elements
